src/java/fr/paris/lutece/plugins/extend/service/DefaultExtendableResourceService.java

```diff
@@ -1,138 +1,151 @@
-     */
-    @Override
-    public String getResourceUrl( String strIdResource, String strResourceType )
-    {
-        return null;
-    }
-}
+/*
+ * Copyright (c) 2002-2021, City of Paris
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *  1. Redistributions of source code must retain the above copyright notice
+ *     and the following disclaimer.
+ *
+ *  2. Redistributions in binary form must reproduce the above copyright notice
+ *     and the following disclaimer in the documentation and/or other materials
+ *     provided with the distribution.
+ *
+ *  3. Neither the name of 'Mairie de Paris' nor 'Lutece' nor the names of its
+ *     contributors may be used to endorse or promote products derived from
+ *     this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * License 1.0
+ */
+package fr.paris.lutece.plugins.extend.service;
+
+import fr.paris.lutece.plugins.extend.business.IDefaultExtendableResourceDAO;
+import fr.paris.lutece.plugins.extend.business.extender.history.IResourceExtenderHistoryDAO;
+import fr.paris.lutece.portal.service.resource.IExtendableResource;
+
+import jakarta.transaction.Transactional;
+
+import java.util.Locale;
+
+import jakarta.inject.Inject;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Named;
+import jakarta.enterprise.inject.spi.CDI;
+
+/**
+ *
+ * Default service that is invoked when other IExtendableResourceService are not invoked. This service stores in database the information of the
+ * {@link IExtendableResource} :
+ * <ul>
+ * <li>id</li>
+ * <li>resourceType</li>
+ * <li>name</li>
+ * </ul>
+ *
+ */
+
+@ApplicationScoped
+@Named( "extend.defaultExtendableResourceService" )
+public class DefaultExtendableResourceService implements IDefaultExtendableResourceService
+{
+    /** The Constant BEAN_SERVICE. */
+    public static final String BEAN_SERVICE = "extend.defaultExtendableResourceService";
+
+    private IDefaultExtendableResourceDAO _extendableResourceDAO = CDI.current( ).select( IDefaultExtendableResourceDAO.class ).get( );
```

Comment:
Inject the bean IDefaultExtendableResourceDAO
@Inject IDefaultExtendableResourceDAO _extendableResourceDAO;